### PR TITLE
[hotfix] Replace System.out.println with logger for better log management

### PR DESCRIPTION
**Pull request by caicancai**

### CODE REVIEW

**Comment by caicancai** [comment]
@snuyanzin If you have time, could you help me take a look at this PR?

**Comment by vahmed-hamdy** [comment]
@caicancai What is the purpose of this PR? I can see the module is intended to be used as a local example of running pubsub with flink, am I correct?

**Comment by caicancai** [comment]
> @caicancai What is the purpose of this PR? I can see the module is intended to be used as a local example of running pubsub with flink, am I correct?

Hello, thank you for your review. 

Yes, your feeling is correct. This is an example test. However, when I started debugging, I found that the producer log is as follows
`                System.out.println("Published message: " + i);`
The consumer log is as follows
```
    private static Integer printAndReturn(Integer i) {
        LOG.info("Processed message with payload: " + i);
        return i;
    }
```

I think it might be better to keep it unified. If it's not suitable, I can close this PR.

File: flink-examples-streaming-gcp-pubsub/src/main/java/org/apache/flink/streaming/examples/gcp/pubsub/PubSubPublisher.java [hunk 1]
```diff
@@ -21,11 +21,14 @@
...
 import com.google.pubsub.v1.PubsubMessage;
 import com.google.pubsub.v1.TopicName;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.math.BigInteger;
 
 /** Helper class to send PubSubMessages to a PubSub topic. */
 class PubSubPublisher {
+    private static final Logger LOG = LoggerFactory.getLogger(PubSubExample.class);
```
Comment (snuyanzin) [inline flink-examples-streaming-gcp-pubsub/src/main/java/org/apache/flink/streaming/examples/gcp/pubsub/PubSubPublisher.java]:
```suggestion
    private static final Logger LOG = LoggerFactory.getLogger(PubSubPublisher.class);
```

Comment (caicancai) [reply flink-examples-streaming-gcp-pubsub/src/main/java/org/apache/flink/streaming/examples/gcp/pubsub/PubSubPublisher.java]: done

**Review by snuyanzin — APPROVED** [review]

**Pull request merged**